weave-js/src/components/PagePanelComponents/Home/Browse3/pages/PlaygroundPage/PlaygroundPage.tsx

```diff
@@ -103,6 +115,16 @@ export const PlaygroundPageInner = (props: PlaygroundPageProps) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [call.loading]);
 
+  useEffect(() => {
+    if (!callWithCosts.loading && callWithCosts.result) {
+      if (callWithCosts.result.traceCall?.inputs) {
+        setPlaygroundStateFromTraceCall(callWithCosts.result.traceCall);
+      }
+    }
+    // Only set the call the first time the page loads, and we get the call
```

Comment:
wait does this mean the query fires multiple times?

Reply:
it does not, but if we add setPlaygroundStateFromTraceCall to the deps, this hook runs a bunch

Reply:
lets memo setPlaygroundStateFromTraceCall then :)

Reply:
if we don't want to do that upstream for some reason we can just make a local memo'd version in this scope

Reply:
fixed

Reply:
it turns out it was memoized already 😝